arbiter/drivers/onedrive.cpp

```diff
@@ -0,0 +1,325 @@
+}
+
+std::string getChildrenEndpoint(const std::string path)
+{
+    return std::string(path + ":/children");
+}
+
+std::string getRefreshUrl()
+{
+    return "https://login.microsoftonline.com/common/oauth2/v2.0/token";
+}
+
+std::vector<char> buildBody(const http::Query& query)
+{
+    const std::string acc(std::accumulate(
+            query.begin(),
+            query.end(),
+            std::string(),
+            [](const std::string& out, const http::Query::value_type& keyVal)
+            {
+                return out + '&' + keyVal.first + '=' + keyVal.second;
+            }));
+    return std::vector<char>(acc.begin(), acc.end());
+}
+
+std::string getQueries(const std::string url)
+{
+    json result;
+
+    //find position of queries in url
+    std::string::size_type pos(url.find("?"));
+    if (pos == std::string::npos)
+    {
+        return result.dump();
+    }
+
+    std::string queries(url.substr(pos + 1));
+
+    do
+    {
+        //get positional values
+        //find next query
+        std::string::size_type nextQueryPos(queries.find_first_of("&"));
+        //if it doesn't exist, go to the end of the string
+        if (nextQueryPos == std::string::npos)
+            nextQueryPos = queries.size();
+        const std::string::size_type separator(queries.find("="));
+
+        //create key-value pair for json
+        const std::string key(queries.substr(0, separator));
+        const std::string value(queries.substr(separator + 1, nextQueryPos - (separator + 1)));
+
+        result.push_back(json::object_t::value_type(key, value));
+
+        /*BREAK HERE*/
+        //was this the last one? if yes, break. if no, resize queries and continue
+        if (nextQueryPos == queries.size())
+            break;
+
+        queries = queries.substr(nextQueryPos + 1);
+
+    } while (true);
+
+    return result.dump();
+}
+
+}
+
+namespace drivers
+{
+
+OneDrive::OneDrive(http::Pool& pool, std::unique_ptr<Auth> auth)
+    : Https(pool)
+    , m_auth(std::move(auth))
+{ }
+
+std::unique_ptr<OneDrive> OneDrive::create(http::Pool& pool, const std::string s)
+{
+    if (auto auth = Auth::create(s))
+    {
+        return makeUnique<OneDrive>(pool, std::move(auth));
+    }
+
+    return std::unique_ptr<OneDrive>();
+}
+
+std::unique_ptr<std::size_t> OneDrive::tryGetSize(const std::string path) const
+{
+    const std::string endpoint(getBaseEndpoint(path));
+    http::Headers headers(m_auth->headers());
+    headers["Content-Type"] = "application/x-www-form-urlencoded";
+    drivers::Https https(m_pool);
+
+    const auto res(https.internalGet(endpoint, headers));
+
+    if (!res.ok())
+    {
+        std::cout <<
+            "Failed get - " << res.code() << ": " << res.str() << std::endl;
+        return std::unique_ptr<std::size_t>();
+    }
+
+    //parse the data into a json object and extract size key value
+    const auto obj = json::parse(res.data());
+    if (obj.find("size") != obj.end())
```

Comment:
`if (obj.count("size"))` is a cleaner check here since the `find` iterator result isn't being used anyway.